avogadro/rendering/solid_first_fs.glsl

```diff
@@ -0,0 +1,97 @@
+//////////////////////////////////////////////////////////////////////
+//
+// First-stage screen-space fragment shader for the solid pipeline
+//
+// It offers ambient occlusion and edge detection capabilities.
+//
+//////////////////////////////////////////////////////////////////////
+
+#version 120
+
+//
+// Input
+//
+
+// texture coordinates
+varying vec2 UV;
+
+//
+// Uniforms
+//
+
+// RGB rendered texture
+uniform sampler2D inRGBTex;
+// Depth rendered texture
+uniform sampler2D inDepthTex;
+
+const float sampleStep = 0.001;
+
+vec3 getNormalAt(vec2 normalUV)
+{
+  float xpos = texture2D(inDepthTex, normalUV + vec2(sampleStep, 0.0)).x;
+  float xneg = texture2D(inDepthTex, normalUV - vec2(sampleStep, 0.0)).x;
+  float ypos = texture2D(inDepthTex, normalUV + vec2(0.0, sampleStep)).x;
+  float yneg = texture2D(inDepthTex, normalUV - vec2(0.0, sampleStep)).x;
+  float xdelta = xpos - xneg;
+  float ydelta = ypos - yneg;
+  vec3 r = vec3(xdelta, ydelta, 2.0 * sampleStep);
+  return normalize(r);
+}
+
+float lerp(float a, float b, float f)
+{
+    return a + f * (b - a);
+}
+
+const vec2 SSAOkernel[16] = vec2[16](
```

Comment:
I'm a little confused by the small magic numbers that are then multiplied by 10.0 in the code?

Wouldn't it be easier to just multiply this all by 10.0 here, e.g.
`vec2(-5.3e-3, 5.0e-3)` etc.?